### PR TITLE
Add dingtalk package

### PR DESCRIPTION
**Pull request by xddxdd**
Official DingTalk Linux client.

Ported from <https://aur.archlinux.org/packages/dingtalk-bin>. Only tested basic messaging functionalities.

### CODE REVIEW

**Comment by Ninlives** [comment]
在我这里不工作，最后一条报错是
```
[0315/175111.296336:ERROR:icu_util.cc(247)] Couldn't mmap icu data file
```

File: packages/dingtalk/default.nix [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    xorg.libXmu
+    xorg.libXrandr
+    xorg.libXrender
+    xorg.libXScrnSaver
+    xorg.libXt
+    xorg.libXtst
+  ];
+in
+stdenv.mkDerivation rec {
+  pname = "dingtalk-resource";
```
Comment (Ninlives) [inline packages/dingtalk/default.nix]: Why not use "dingtalk" as the pname? Users can also use `nix run` to use this derivation directly if the pname matches the binary name.

**Comment by Ninlives** [comment]
I just made some changes to make it work on my site, could you please check the latest commit?

**Comment by xddxdd** [comment]
Confirm that the latest commit works for me. Seems I missed a few libraries in subfolders.

Package name has been fixed.

**Comment by Ninlives** [comment]
Thanks!

**Pull request merged**